### ISSUE TITLE
[macOS] Detect /usr as a wrong java home

### ISSUE THREAD

**Issue by testforstephen** [Eskibear/node-jdk-utils#15]
See https://github.com/redhat-developer/vscode-java/issues/3287#issuecomment-1717525516, on macOS it detects `/usr` as java home by mistake.

> npm i jdk-utils && npx jdk-utils

```added 1 package, and audited 2 packages in 1s

found 0 vulnerabilities
Listing JDKs: 76.725ms
Found:  4
[
  {
    homedir: '/usr/local/Cellar/openjdk@17/17.0.8.1/libexec/openjdk.jdk/Contents/Home',
    hasJavac: true,
    version: { java_version: '17.0.8.1', major: 17 }
  },
  {
    homedir: '/usr/local/Cellar/openjdk@11/11.0.20.1/libexec/openjdk.jdk/Contents/Home',
    hasJavac: true,
    version: { java_version: '11.0.20.1', major: 11 }
  },
  {
    homedir: '/usr/local/Cellar/openjdk/20.0.2/libexec/openjdk.jdk/Contents/Home',
    hasJavac: true,
    version: { java_version: '20.0.2', major: 20 }
  },
  {
    homedir: '/usr',
    isInPathEnv: true,
    hasJavac: true,
    version: { java_version: '17.0.8.1', major: 17 }
  }
]
```

**Comment by Eskibear** [comment]
The most recent change is related to PATH env. Probably due to that. You can try with previous version to verify if it’s the regression. Maybe we should add check for symbol links and returns only real path.

**Comment by testforstephen** [comment]
0.4.6 works, it starts breaking since 0.5.0

**Comment by testforstephen** [comment]
On the latest macOS versions (such as Ventura), `/usr/bin/java` is not a symbolic link to a specific Java executable file, but rather a proxy program that can automatically select and run a suitable JDK from the system when `/usr/bin/java` is invoked. It's a system shortcut in macOS and cannot be used as Java home for other applications. Therefore, we should exclude `/usr` from Java home list when detecting JDK on macOS.

**Comment by Eskibear** [comment]
fixed in v0.5.1
